app/scripts/modules/core/account/account.service.js

```diff
@@ -94,6 +94,42 @@ module.exports = angular.module('spinnaker.core.account.service', [
       return deferred.promise;
     });
 
+    let getUniqueRegionsForAllAccounts = _.memoize((provider) => {
+      return getRegionsKeyedByAccount(provider)
+        .then(function(regionsByAccount) {
+          let regions = _.chain(regionsByAccount)
+            .values()
+            .map(acct => acct.regions)
+            .flatten()
+            .map(reg => reg.name)
+            .uniq()
+            .value();
+
+          return regions;
+        });
+    });
+
+    let getUniqueGceZonesForAllAccounts = _.memoize((provider) => {
+      return getRegionsKeyedByAccount(provider)
+        .then(function(regionsByAccount) {
+          return _.chain(regionsByAccount)
+            .values()
+            .map(acct => acct.regions)
+            .flatten()
+            .reduce((acc, obj) => {
+              Object.keys(obj).forEach((key) => {
+                if(acc[key]) {
+                  acc[key] = _.uniq(acc[key].concat(obj[key]));
+                } else {
+                  acc[key] = obj[key];
+                }
+              });
+              return acc;
+            }, {})
+            .value();
+        });
+    });
```

Comment:
I think this makes sense for now, but we should consider a different approach longer-term - either:
  1. change the backend (either in gate or clouddriver, if gate is just acting as a passthrough here) so that all providers returns credentials in a similar shape, or
  2. allow providers to specify an override for a `getUniqueRegionsForAllAccount` method, kind of like we do with search result formatting (https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/search/infrastructure/infrastructureSearch.service.js#L90-L101)

I would lean towards (1), but there may be good reasons why the credentials for GCE have a different shape than the credentials for AWS - defer to @duftler on that.